scripts/ember_customization.py

```diff
@@ -161,3 +161,82 @@ def apply_custom_pf_constraint(n,
     # 7) enforce band/cap
     m.add_constraints(energy >= E_min, name=f"{link_name}_annual_min")
     m.add_constraints(energy <= E_max, name=f"{link_name}_annual_max")
+    
+    
+def include_coal_chps_for_selected_countries(n, costs, CHP_ppl_fn):
+    country_code_map = {'Poland': 'PL', 'Czechia': 'CZ', 'Greece': 'GR', 'Germany': 'DE'}
+    focus_full= country_code_map.keys()
+    df = pd.read_excel(CHP_ppl_fn, sheet_name='coal_chp')
+    df = df.query("bus in @focus_full")
+    carrier_mapping = {'Hard coal': 'coal', 'Lignite': 'lignite'}
```

Comment:
as discussed earlier, we'll need to add `gas` here. Not sure how the mapping would be -- depends, probably, on what the input data lists for gas CHPs